classes/asserters/blackfire.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+namespace mageekguy\atoum\blackfire\asserters;
+
+use Blackfire\Client;
+use Blackfire\Profile\Configuration;
+use Blackfire\Exception\ExceptionInterface;
+use Blackfire\Profile;
+use mageekguy\atoum\asserter;
+use mageekguy\atoum\locale;
+use mageekguy\atoum\tools\variable;
+
+class blackfire extends asserter
+{
+    /**
+     * @var Configuration
+     */
+    protected $configuration;
+
+    /**
+     * @var Client|null
+     */
+    protected $client;
+
+    /**
+     * @param asserter\generator $generator
+     * @param variable\analyzer $analyzer
+     * @param locale $locale
+     */
+    public function __construct(asserter\generator $generator = null, variable\analyzer $analyzer = null, locale $locale = null)
+    {
+        parent::__construct($generator, $analyzer, $locale);
+        $this->configuration = new Configuration();
+    }
+
+
+    /**
+     * @param string $assertion
+     * @param string $name
+     *
+     * @return $this
+     */
+    public function assert($assertion, $name = '')
+    {
+        $this->configuration->assert($assertion, $name);
+
+        return $this;
+    }
+
+    /**
+     * @param Configuration $configuration
+     *
+     * @return $this
+     */
+    public function setConfiguration(Configuration $configuration)
+    {
+        $this->configuration = $configuration;
+
+        return $this;
+    }
+
+    /**
+     * @return Configuration
+     */
+    public function getConfiguration()
+    {
+        return $this->configuration;
+    }
+
+    /**
+     * @param Client $client
+     *
+     * @return $this
+     */
+    public function setClient(Client $client)
+    {
+        $this->client = $client;
+
+        return $this;
+    }
+
+    /**
+     * @throws \LogicException
+     *
+     * @return Client|null
+     */
+    public function getClient()
+    {
+        if (null === $this->client) {
+            throw new \LogicException('Client has not been set');
+        }
+        return $this->client;
+    }
+
+    /**
+     * @param \closure $callback
+     *
+     * @return \mageekguy\atoum\asserters\object
+     */
+    public function profile($callback)
+    {
+        $client = $this->getClient();
+
+        if (!($callback instanceof \closure))
+        {
+            $this->fail($this->_('profile parameter is not a closure'));
```

Comment:
or if you want to stick to a check here (with the `if`)  please allow any callable (using `is_callable`)